rosco-manifests/src/test/java/com/netflix/spinnaker/rosco/manifests/helm/HelmSetArgumentParserTest.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2024 Salesforce, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.rosco.manifests.helm;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Tests for HelmSetArgumentParser, ensuring it correctly parses string inputs into structured maps.
+ * This class includes tests for various input scenarios, including basic key-value pairs, nested
+ * structures, lists, type inference, and error handling. The tests are inspired by Helm's own
+ * parser tests, adapted to validate the Java implementation's compatibility with expected Helm
+ * behavior. Reference to Helm's Go parser tests for comparison and validation: <a
+ * href="https://github.com/helm/helm/blob/v3.14.2/pkg/strvals/parser_test.go">Helm Go Parser
+ * Tests</a>.
+ */
+public class HelmSetArgumentParserTest {
+
+  private static Stream<Arguments> helmParseArgs() {
+
+    return Stream.of(
+        Arguments.of("name1=value1", Map.of("name1", "value1"), true),
```

Comment:
Can you add a float test?

Reply:
Done in https://github.com/spinnaker/rosco/pull/1135/commits/a0da916e0c79f8399d65a7d51ee7c21e81c9508c.

Reply:
Turns out things are more complicated than we originally thought...